avocado/utils/vmimage.py

```
@@ -633,6 +633,109 @@ def _take_snapshot(self):
+                return False
+
+            if version and str(metadata.get("version", "")) != str(version):
+                return False
+
+            if build and str(metadata.get("build", "")) != str(build):
+                return False
+
+            if compatible_arches and metadata.get("arch") not in compatible_arches:
+                return False
+
+            return True
```

Ruff (0.12.2)

Return the negated condition directly

Inline condition

(SIM103)